### PR TITLE
Ddt 957 implement versioning get booking

### PR DESCRIPTION
**Pull request by pimg**
resolves DDT-957

### CODE REVIEW

File: src/main/java/org/dcsa/bkg/service/impl/BKGServiceImpl.java [hunk 1]
```diff
@@ -240,28 +239,23 @@ private Mono<List<CommodityTO>> createCommoditiesByBookingIDAndTOs(
...
+    return Flux.fromIterable(valueAddedServiceRequests)
+        .map(
+            valueAddedServiceRequestTO -> {
+              ValueAddedServiceRequest valueAddedServiceRequest = new ValueAddedServiceRequest();
+              valueAddedServiceRequest.setBookingID(bookingID);
+              valueAddedServiceRequest.setValueAddedServiceCode(
+                  valueAddedServiceRequestTO.getValueAddedServiceCode());
+              return valueAddedServiceRequest;
+            })
+        .flatMap(valueAddedServiceRequestRepository::save)
```
Comment (nt-as) [inline src/main/java/org/dcsa/bkg/service/impl/BKGServiceImpl.java:258]: I would probably use `.concatMap` here to preserve ordering.  It probably only matters if this is also echo'ed in the output though.

File: src/main/java/org/dcsa/bkg/service/impl/BKGServiceImpl.java [hunk 2]
```diff
@@ -280,24 +274,21 @@ private Mono<List<ReferenceTO>> createReferencesByBookingIDAndTOs(
...
         .map(
-            r -> {
+            referenceTO -> {
+              Reference reference = new Reference();
+              reference.setBookingID(bookingID);
+              reference.setReferenceType(referenceTO.getReferenceType());
+              reference.setReferenceValue(referenceTO.getReferenceValue());
+              return reference;
+            })
+        .flatMap(referenceRepository::save)
```
Comment (nt-as) [inline src/main/java/org/dcsa/bkg/service/impl/BKGServiceImpl.java:286]: Ditto (`.concatMap`)

File: src/main/java/org/dcsa/bkg/service/impl/BKGServiceImpl.java (hunk 1 above)
Comment (pimg) [reply src/main/java/org/dcsa/bkg/service/impl/BKGServiceImpl.java]: changed accordingly

File: src/main/java/org/dcsa/bkg/service/impl/BKGServiceImpl.java (hunk 2 above)
Comment (pimg) [reply src/main/java/org/dcsa/bkg/service/impl/BKGServiceImpl.java]: changed accordingly

**Review by nt-as — APPROVED** [review]

**Pull request merged**